src/v/kafka/requests/schemata/generator.py

```diff
@@ -715,7 +715,7 @@ class response;
 {{- struct_serde(struct, field_decoder) | indent }}
 }
 {%- else %}
-void {{ struct.name }}::decode(iobuf buf, api_version version) {
+void {{ struct.name }}::decode(iobuf buf, [[maybe_unused]] api_version version) {
```

Comment:
iv you remove the `version` from the `api_version version` you don't need to tag the argument as unused.

Reply:
this is a generator code and in some cases it generates code which uses `version` and in another which doesn't (depending on a schema). init_producer_id is the first schema which doesn't use `version` so I have to add the attribute to come the most generic case :-(

Reply:
yeh, i think they are all maybe_unused now. you found the last one!